Modules/_pickle.c

```diff
@@ -5988,12 +5995,20 @@ load_stack_global(PickleState *st, UnpicklerObject *self)
     PyObject *global_name;
 
     PDATA_POP(st, self->stack, global_name);
+    if (global_name == NULL) {
```

Comment:
Actually, this change was not necessary. It is safe to call few PDATA_POPs sequentially and check for error later.

Reply:
Yes, but it only calls `PyErr_SetString()` which is safe to be called with an error set.

Reply:
Ah, that is true. However, that is an implementation detail of `PDATA_POP`. If that changes in the future, we should be safer now.

Reply:
I do not say that this change was wrong. It was just not necessary with the current code.

When improving error handling in this code several years ago I considered options of raising an exception explicitly outside of PDATA_POP (after checking its result) and implicitly inside of it, and choose the latter, because it led to smaller code. But the difference was small, so I could chose the other option, and PDATA_POP could not use any C API at all. It is just useless historical reference.

Reply:
Thanks for the insight!